api/src/pcapi/core/educational/api/offer.py

```diff
@@ -604,6 +607,125 @@ def attach_image(
         db.session.commit()
 
 
+def update_collective_offer(
+    offer_id: int,
+    new_values: dict,
+) -> None:
+    offer_to_update = educational_models.CollectiveOffer.query.filter(
+        educational_models.CollectiveOffer.id == offer_id
+    ).first()
+
+    educational_validation.check_if_offer_not_used_or_reimbursed(offer_to_update)
+
+    if "venueId" in new_values and new_values["venueId"] != offer_to_update.venueId:
+        offerer = repository.get_by_collective_offer_id(offer_to_update.id)
+        new_venue = repository.get_venue_by_id(new_values["venueId"])
+        if not new_venue:
+            raise offers_exception.VenueIdDontExist()
+        if new_venue.managingOffererId != offerer.id:
+            raise offers_exception.OffererOfVenueDontMatchOfferer()
+
+    updated_fields = _update_collective_offer(offer=offer_to_update, new_values=new_values)
+
+    search.async_index_collective_offer_ids([offer_to_update.id])
+
+    notify_educational_redactor_on_collective_offer_or_stock_edit(
+        offer_to_update.id,
+        updated_fields,
+    )
+
+
+def update_collective_offer_template(offer_id: int, new_values: dict) -> None:
+
+    query = educational_models.CollectiveOfferTemplate.query
+    query = query.filter(educational_models.CollectiveOfferTemplate.id == offer_id)
+    offer_to_update = query.first()
+
+    if "venueId" in new_values and new_values["venueId"] != offer_to_update.venueId:
+        offerer = repository.get_by_collective_offer_id(offer_to_update.id)
+        new_venue = repository.get_venue_by_id(new_values["venueId"])
+        if not new_venue:
+            raise offers_exception.VenueIdDontExist()
+        if new_venue.managingOffererId != offerer.id:
+            raise offers_exception.OffererOfVenueDontMatchOfferer()
```

Comment:
Une petite optimisation au cas ou la venue n'existe pas

```suggestion
        new_venue = repository.get_venue_by_id(new_values["venueId"])
        if not new_venue:
            raise offers_exception.VenueIdDontExist()
        offerer = repository.get_by_collective_offer_id(offer_to_update.id)
        if new_venue.managingOffererId != offerer.id:
            raise offers_exception.OffererOfVenueDontMatchOfferer()
```